lib/rubocop/cop/rspec/focus.rb

```diff
@@ -89,9 +89,11 @@ def on_send(node)
         private
 
         def focus_metadata(node, &block)
-          yield(node) if focused_block?(node)
-
-          metadata(node, &block)
+          if focused_block?(node)
```

Comment:
What if inline this method? Won't we be able to remove one "if" statement?

Reply:
`focused_block?` is a node matcher; I’m not sure how to inline it?

Reply:
```ruby
        def on_send(node)
          return if node.chained? || node.each_ancestor(:def, :defs).any?

          if focused_block?(node)
            add_offense(node) do |corrector|
              correct_send(corrector, node)
            end
          else
            metadata(node) do |focus|
              add_offense(focus) do |corrector|
                corrector.remove(with_surrounding(focus))
              end
            end
          end
        end
```

Reply:
This causes an offence, though `Metrics/MethodLength: Method has too many lines. [12/10]`.

To me, this improves the separation of the two cases:
 - fit/fdescribe/... focused example/example groups methods
 - examples/example groups focused with `focus` metadata

It just happens that the entry point to detect them is the same, `on_send`.

For those two cases:
1. node pattern to detect is different
2. the selection for offence is different
3. the correction is different

Reply:
Brilliant. Thanks.